pkg/gencosmosdb/cosmosdb/cosmosdb.go

```diff
@@ -24,6 +24,7 @@ type Options struct {
 	PreTriggers         []string
 	PostTriggers        []string
 	PartitionKeyRangeID string
+	Continuation        string
```

Comment:
Not sure if it makes sense to have it in `Options` as we use it only in iterators. We don't need it in `Create`, `Get`, `Replace`, `Delete`.

Reply:
I think Continuation is OK; I wonder whether it would be clearer to initialize i.continuation from options.Continuation in each call that returns an iterator rather than doing `else if i.options != nil && i.options.Continuation != ""`.  Maybe doing so clarifies a little that options.Continuation is the initializing continuation and never the dynamic continuation.

MaxItemCount feels more problematic, for different reasons.  It feels much more like a parameter to Next().  At the moment your PR will start setting the X-Ms-Max-Item-Count header on any call which calls setOptions.  I'd be tempted to change the signature of Next() to receive the max item count as an int, and have people send in -1 if they don't care.

Reply:
> I wonder whether it would be clearer to initialize i.continuation from options.Continuation in each call

It still not super clean because of `options != nil` check, but it's easier to understand that we only set it once this way.

Re: `MaxItemCount`. I like idea of passing it into `Next`.